### PR TITLE
SNOW-955536: Multiple SAML Integration

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-lf**
### Description
Add multiple SAML Integration

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

File: Snowflake.Data/Core/Authenticator/ExternalBrowserAuthenticator.cs [hunk 1]
```diff
@@ -187,7 +206,7 @@ private static void StartBrowser(string url)
             // The following code is learnt from https://brockallen.com/2016/09/24/process-start-for-urls-on-net-core/
 #if NETFRAMEWORK
             // .net standard would pass here
-            Process.Start(url);
+            Process.Start(new ProcessStartInfo(url) { UseShellExecute = true });
```
Comment (sfc-gh-dstempniak) [inline Snowflake.Data/Core/Authenticator/ExternalBrowserAuthenticator.cs:209]: why this change?

File: Snowflake.Data.Tests/UnitTests/SFSessionPropertyTest.cs [hunk 2]
```diff
@@ -379,6 +420,8 @@ public static IEnumerable<TestCase> ConnectionStringTestCases()
                 testCaseWithFileTransferMaxBytesInMemory,
                 testCaseWithIncludeRetryReason,
                 testCaseWithDisableQueryContextCache,
+                testCaseWithDisableConsoleLogin,
+                testCaseComplicatedAccountName,
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/UnitTests/SFSessionPropertyTest.cs:424]: Will remove

File: Snowflake.Data/Core/Authenticator/ExternalBrowserAuthenticator.cs (hunk 1 above)
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data/Core/Authenticator/ExternalBrowserAuthenticator.cs:209]: I encountered the error `System.ComponentModel.Win32Exception: The system cannot find the file specified` while I was testing locally. And setting the `UseShellExecute` property to true fixed the error

**Review by sfc-gh-dstempniak — APPROVED** [review]

**Review by sfc-gh-knozderko — APPROVED** [review]
